### DISCUSSION TITLE
Solved: Unable to activate ST7789 using STM32F103

### DISCUSSION THREAD

**Discussion by puby** [Bodmer/TFT_eSPI#2826] [Compatible displays and setup files]
I tried modifying User_Setup.h with the following content, but it doesn't work, can anyone assist me?

#define STM32

#define ST7789_DRIVER   // Full configuration option, define additional parameters below for this display
#define TFT_HEIGHT 320  // ST7789 240 x 320

#define TFT_SPI_PORT 1 // SPI port 1 maximum clock rate is 55MHz
#define TFT_MOSI PA7   // SDA 
#define TFT_MISO PA6   // <---- ST7789 does not have this pin
#define TFT_SCLK PA5

#define TFT_CS   PA2 // Chip select control pin to TFT CS  (PB4)
#define TFT_DC   PA3 // Data Command control pin to TFT DC (may be labelled RS = Register Select) (PB10)
#define TFT_RST  PA4 // Reset pin to TFT RST (or RESET) (PA8)

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:-.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts
#define SMOOTH_FONT

#define SPI_FREQUENCY  55000000 // STM32 SPI1 only (SPI2 maximum is 27MHz)
#define SPI_READ_FREQUENCY  20000000
#define SPI_TOUCH_FREQUENCY  2500000

Also, does the content of the file User_Setup_Select.h need to be changed? I didn't find the configuration file for STM32F103 with ST7789.

Thank.

**Comment by Bodmer** [comment]
Setup 32 could be used as a starting point if you have the "Blue pill" style board.

**Reply by puby** [reply]
Hi Bodmer,

Thank you very much for your kind assistance.

My problem is solved
In User_Setup_Select.h, just MARK the line //#include <User_Setups/Setup32_ILI9341_STM32F103.h> // Setup for "Blue/Black Pill".

Excellent library, I love it so much, thank you again for your dedication.
